### PR TITLE
Create resources

### PR DESCRIPTION
**Pull request by DenizCankurtaran**
Each component has a modal which can create the corresponding resource.
Pods
Screenshot from 2021-04-06 13-36-59

Deployments
Screenshot from 2021-04-06 13-37-13

PVCs
Screenshot from 2021-04-06 13-37-39

Services
Screenshot from 2021-04-06 13-37-51

Ingresses
Screenshot from 2021-04-06 13-38-15

Example of deployment
1.
Screenshot from 2021-04-06 14-10-55
2.
Screenshot from 2021-04-06 14-11-25

Every modal is dynamic and allows multiple values.
e.g.
image

### CODE REVIEW

**Comment by troeger** [comment]
I am leaving the code review to @kat-hi , and focus on the screenshots here:

- I like the "+" optics, so that the novice user is not overwhelmed with form options.
- [ ] The buttons should be right-aligned. The green button should be on the border.
- [ ] Deployment: I would drop the "template" term and just talk about a pod here, for simplification.
- [ ] Deployment: It would be nice if the label fields would be connected, so if I fill out the match label, the pod labels are automatically filled with these values.
- [ ] PVC: The access mode should be a dropdown list, the possible values are pre-defined.
- [ ] PVC: The storage class should be a dropdown list. Please create an according API issue for getting the list of available storage classes.
- [ ] PVC: Size is a major pain point, due to the different supported metrics. Maybe there is a cool JS widget for making "data amount" choices user friendly.

Some alignments look strange, but you see this by yourself. Very nice!

**Comment by DenizCankurtaran** [comment]
Attempt to make PVC size more user friendly, other size amounts can easily be added
image

File: src/components/Container/DeploymentModal.vue [hunk 1]
```diff
@@ -29,23 +119,39 @@
...
+      rules: {
+        required: value => !!value || 'Required.'
+      }
+
     }
   },
   methods: {
     async post_deployment (e) {
       e.preventDefault()
-      let response = await backend.post(`/deployments/${this.namespace}/`, {
```
[the remark this answers is not in the document]
Comment (DenizCankurtaran) [reply src/components/Container/DeploymentModal.vue:45]: I do, I have refactored this a while ago.
[current approach](https://github.com/kubeportal/kubeportal-frontend/pull/109/files#diff-7c2301c4fbad584303842579970936d20aedef48b0c483e7ceba28f3cab80d7aR154)

**Review by troeger — APPROVED** [review]
I think the changes so far are enough for Alpha. Please create according new issues for the remaining "nice to have" stuff.

**Pull request merged**